### PR TITLE
Adding clarification to RHL3 docs + JSX highlight in code blocks

### PR DESCRIPTION
**Pull request by tsaiDavid**
@gadicc @hedgerh @gaearon - Went ahead and added the changes I mentioned along with some other little touches.

- Clarification and showing the JSX syntax with the AppContainer component wrapping RootContainer - hopefully this is more beginner friendly and less confusing since v3 will require a bit more config.
- Switched from `js` to `jsx` for the code blocks, if I'm not mistaken it's GitHub markdown friendly 👍 
- Provided super basic start for us to show usage within webpack loader (just to get the ball rolling)

### CODE REVIEW

**Comment by tsaiDavid** [comment]
Just want to point out I was also able to update the shell command example, prefixed it with the `$` symbol that we typically see, indicating that it is a command to be used from one's terminal.

An example was also not in consistent ES6 as in our other examples, so I updated it.

**Comment by tsaiDavid** [comment]
@gadicc #257 #243 - apologies for all the comments, just want to make sure I'm referencing the issue correctly, please let me know if there's anything else I can do.

**Comment by gadicc** [comment]
this is awesome, @tsaiDavid, thanks!  all very welcome improvements.  just one issue which I'll comment on in the diff and then I'm happy to merge.  yeah, for future it's enough to just put a #257 in there somewhere and we'll get a notification (I think :)).

File: docs/README.md [hunk 1]
```diff
@@ -20,7 +20,8 @@ First we had React Hot Loader, which was superseded by React Transform.  Each ti
 *   Works with or without Babel
 *   Everything you need in a single repo
 
-For more information on the evolution of approaches taken, see [](https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf)https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf.
+For more information on the evolution of approaches taken
```
Comment (gadicc) [inline docs/README.md]: just this snuck through :)

**Comment by gadicc** [comment]
@tsaiDavid I'm going to fix that line on my side, these were actually intended for the gh-pages (sorry that's my bad) so need to move these there and then I'll ping you again.  Thanks!

**Pull request merged**